portfolio/src/main/webapp/script.js

```diff
@@ -50,16 +50,28 @@ function showPics() {
 }
 
 /* Gets array list of messages from servlet, converts to JSON, and displays it on home page. */
-function getJSONMessages() {
+function getBucketList() {
   fetch('/data').then(response => response.json()).then((list) => {
     const listElement = document.getElementById('list-container');
     listElement.innerHTML = '';
     let bucketList = "";
 
-    for (let i = 0; i < list.length; i++) {
-      bucketList += (i + 1) + ": " + list[i] + '\n';
+    // Display elements in a numerical list
+    let count = 1;
+    for (i = 0; i < list.length; i++) {
+      if (list[i] === "") {
```

Comment:
This can be simplified to `if (!list[i]) {` because an empty string will evaluate to false, but any other string will evaluate to true. On the other hand, I get it if you'd rather keep the current version as it's more explicit. Do whichever you think strikes the best balance of clarity and simplicity.

See below for IMO an even simpler approach that uses `if (list[i])` without the negation.